potentialventures/cocotb
# AvalonMemory read burst capability

This pull request add read burst capability for AvalonMemory class. It's usefull to test avalon-mm master read capabilities.
A test "test_avalon" has been added in tests directory. To test it, the altera template : 
https://www.altera.com/support/support-resources/design-examples/intellectual-property/embedded/nios-ii/exm-avalon-mm.html

Has been used and modified to include a verilog model of scfifo.

To test, go to tests directory and do: 
```bash
make test_avalon
```

A gtkwave configuration file is also available to see signals in gtkwave 
tests/test_cases/test_avalon/test_avalon_cfg.gtkw
